File: TwitterClient/src/protocols/STOMP/frames/ConnectFrame.cpp
/*
 * ConnectFrame.cpp
 *
 *  Created on: Jan 7, 2014
 *      Author: amio
 */

#include "ConnectFrame.h"


//ConnectFrame::ConnectFrame()
//{
//}

ConnectFrame::ConnectFrame(const string& hostIP, unsigned short port,
		const string& username, const string& password):_hostIP(hostIP),_port(port),_username(username),_password(password) {

}

ConnectFrame::~ConnectFrame() {
}

string ConnectFrame::toString() {
	vector<pair<string,string> > vec;
	vec.push_back(make_pair("accept-version","1.2"));
	vec.push_back(make_pair("host",_hostIP));
	return makeFrame("CONNECT",vec,"");
}

File: ReactorServer/src/spl/server/stomp/frames/ConnectFrame.java
package spl.server.stomp.frames;

import java.util.HashMap;
import java.util.Map;



public class ConnectFrame extends StompFrame {
	private String _hostIP;
	private String _username;
	private String _password;
	
	public ConnectFrame(String version,String host, String username,String password) {
		_hostIP = host;
		_username = username;
		_password = password;
	}


	@Override
	public String getEncodedString() {
		Map<String, String> map = new HashMap<>();
		map.put("accept-version","1.2");
		map.put("host",_hostIP);
		map.put("login",_username);
		map.put("passcode",_password);

		return makeFrame("CONNECT", map, "");
	}

	public String getPassword() {
		return _password;
	}
	public String getUsername() {
		return _username;
	}

}
